Adding details for getting started in the documentations (#558)
ZIB-IOL/FrankWolfe.jl: README.md

## Getting started

Let's say we want to solve the following minimization problem 
```math
\min_{p \in  Δ(n)} p_1^2 + \dots + p_n^2,
```
where $`Δ(n)= \{p \in R^n_{\geq 0} | p_1 + \dots + p_n =1\}`$ is the _probability simplex_.

Using `FrankWolfe.jl`, let's write a minimal code solving this problem in dimension $n=3$.
The main function is **`FrankWolfe.frank_wolfe`** and it requires: 

* a **function `f`** that computes the values of the objective function $f$;
* a **function `grad!`** that computes in-place the gradient of the objective function $f$;
* a **subtype of `FrankWolfe.LinearMinimizationOracle`** for which a method of        `FrankWolfe.compute_extreme_point` has been implemented (see [here](https://zib-iol.github.io/FrankWolfe.jl/dev/basics/#Linear-Minimization-Oracles));
* a **starting vector `p0`**.

```julia
julia> using FrankWolfe

> # #objective function f(p) = p_1^2 + ... + p_n^2

Reviewer: ```suggestion
# objective function f(p) = p_1^2 + ... + p_n^2
```
not sure if the comment is necessary, it seems redundant with the code

julia> f(p) = sum(abs2, p)

# #in-place gradient computation for f thanks to '.='
julia> grad!(storage, p) = storage .= 2p  

> # #pre-defined type for which a method computes a solution of min ⟨p,d⟩ st. p ∈ Δ 

Reviewer: ```suggestion
# pre-defined type implementing the linear minimization oracle interface for the simplex
```

 0.3333333286623478
```

> **Note** that active-set based methods like Away Frank-Wolfe and Blended Pairwise Conditional Gradient also include a post processing step. 

Reviewer: ```suggestion
**Note** that active-set based methods like the Away-step Frank-Wolfe and Blended Pairwise Conditional Gradients also include a post-processing step. 
```
